diff --git a/ctl_equal.c b/ctl_equal.c
index 1127a27..d43e298 100644
--- a/ctl_equal.c
+++ b/ctl_equal.c
@@ -24,6 +24,14 @@
 #include "ladspa.h"
 #include "ladspa_utils.h"
 
+#ifndef CAPS_SO
+#define CAPS_SO_PATH "/usr/lib/ladspa/caps.so"
+#else
+#define STRINGIZE(x) #x
+#define STRINGIZE_CAPS(x) STRINGIZE(x)
+#define CAPS_SO_PATH STRINGIZE_CAPS(CAPS_SO)
+#endif
+
 typedef struct snd_ctl_equal_control {
 	long min;
 	long max;
@@ -166,7 +174,7 @@ SND_CTL_PLUGIN_DEFINE_FUNC(equal)
 	snd_config_iterator_t it, next;
 	snd_ctl_equal_t *equal;
 	const char *controls = ".alsaequal.bin";
-	const char *library = "/usr/lib/ladspa/caps.so";
+	const char *library = CAPS_SO_PATH;
 	const char *module = "Eq10";
 	long channels = 2;
 	const char *sufix = " Playback Volume";
diff --git a/pcm_equal.c b/pcm_equal.c
index b0b4265..fdff9b3 100644
--- a/pcm_equal.c
+++ b/pcm_equal.c
@@ -27,6 +27,14 @@
 #include "ladspa.h"
 #include "ladspa_utils.h"
 
+#ifndef CAPS_SO
+#define CAPS_SO_PATH "/usr/lib/ladspa/caps.so"
+#else
+#define STRINGIZE(x) #x
+#define STRINGIZE_CAPS(x) STRINGIZE(x)
+#define CAPS_SO_PATH STRINGIZE_CAPS(CAPS_SO)
+#endif
+
 typedef struct snd_pcm_equal {
 	snd_pcm_extplug_t ext;
 	void *library;
@@ -150,7 +158,7 @@ SND_PCM_PLUGIN_DEFINE_FUNC(equal)
 	snd_pcm_equal_t *equal;
 	snd_config_t *sconf = NULL;
 	const char *controls = ".alsaequal.bin";
-	const char *library = "/usr/lib/ladspa/caps.so";
+	const char *library = CAPS_SO_PATH;
 	const char *module = "Eq10";
 	long channels = 2;
 	int err;
