--- common/lib/modules/fglrx/build_mod/firegl_public.c	2015-05-04 03:57:25.000000000 -0400
+++ common/lib/modules/fglrx/build_mod/firegl_public.c.new	2015-05-04 03:58:14.000000000 -0400
@@ -269,7 +269,7 @@
 #endif
 
 #ifdef MODULE_LICENSE
-MODULE_LICENSE("Proprietary. (C) 2002 - ATI Technologies, Starnberg, GERMANY");
+MODULE_LICENSE("GPL");
 #endif
 #ifdef MODULE_DEVICE_TABLE
 MODULE_DEVICE_TABLE(pci, fglrx_pci_table);
