--- a/vmci-only/linux/vmciKernelIf.c	2014-06-12 19:20:45.000000000 -0400
+++ b/vmci-only/linux/vmciKernelIf.c	2015-05-13 21:36:39.000000000 -0400
@@ -1227,9 +1227,11 @@
       if (isIovec) {
          struct iovec *iov = (struct iovec *)src;
          int err;
+         struct iov_iter from;
 
+	 iov_iter_init(&from, READ, &iov, 1, iov->iov_len);
          /* The iovec will track bytesCopied internally. */
-         err = memcpy_fromiovec((uint8 *)va + pageOffset, iov, toCopy);
+	 err = (copy_from_iter((uint8*)va + pageOffset, toCopy, &from) != toCopy);
          if (err != 0) {
             if (!kernelIf->isDataMapped) {
                kunmap(kernelIf->page[pageIndex]);
@@ -1302,9 +1304,11 @@
       if (isIovec) {
          struct iovec *iov = (struct iovec *)dest;
          int err;
+	 struct iov_iter to;
 
+	 iov_iter_init(&to, WRITE, &iov, 1, iov->iov_len);
          /* The iovec will track bytesCopied internally. */
-         err = memcpy_toiovec(iov, (uint8 *)va + pageOffset, toCopy);
+	 err = (copy_to_iter((uint8*)va + pageOffset, toCopy, &to) != toCopy);
          if (err != 0) {
             if (!kernelIf->isDataMapped) {
                kunmap(kernelIf->page[pageIndex]);
