--- a/vmblock-only/linux/dentry.c	2014-06-12 19:20:45.000000000 -0400
+++ b/vmblock-only/linux/dentry.c	2015-05-13 18:29:15.000000000 -0400
@@ -62,6 +62,20 @@
 DentryOpRevalidate(struct dentry *dentry,  // IN: dentry revalidating
                    struct nameidata *nd)   // IN: lookup flags & intent
 {
+   #if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 19, 0)
+   struct nameidata {
+          struct path     path;
+          struct qstr     last;
+          struct path     root;
+          struct inode    *inode; /* path.dentry.d_inode */
+          unsigned int    flags;
+          unsigned        seq, m_seq;
+          int             last_type;
+          unsigned        depth;
+          char *saved_names[MAX_NESTED_LINKS + 1];
+   };
+   #endif
+
    VMBlockInodeInfo *iinfo;
    struct nameidata actualNd;
    struct dentry *actualDentry;
