--- a/vmci-only/linux/driver.c	2014-06-12 19:20:45.000000000 -0400
+++ b/vmci-only/linux/driver.c	2015-05-13 20:05:55.000000000 -0400
@@ -737,7 +737,7 @@
          goto init_release;
       }
 
-      user = current_uid();
+      user = current_uid().val;
       retval = VMCIContext_InitContext(initBlock.cid, initBlock.flags,
                                        0 /* Unused */, vmciLinux->userVersion,
                                        &user, &vmciLinux->context);
