diff -rupN a/debuggers/gdb/gdb.cpp b/debuggers/gdb/gdb.cpp
--- a/debuggers/gdb/gdb.cpp	2015-02-02 09:19:46.000000000 -0500
+++ b/debuggers/gdb/gdb.cpp	2015-08-03 18:15:43.000000000 -0400
@@ -44,7 +44,10 @@
 using namespace GDBDebugger;
 
 GDB::GDB(QObject* parent)
-: QObject(parent), process_(0), sawPrompt_(false), currentCmd_(0), receivedReply_(false), isRunning_(false), childPid_(0)
+: QObject(parent), process_(0), sawPrompt_(false), currentCmd_(0), receivedReply_(false), isRunning_(false)
+#if defined(Q_OS_MAC)
+, childPid_(0)
+#endif
 {
 }
 
@@ -157,9 +160,12 @@ void GDB::interrupt()
 {
     //TODO:win32 Porting needed
     int pid = process_->pid();
+    #if defined(Q_OS_MAC)
     if (childPid_) {
         ::kill(childPid_, SIGINT);
-    } else if (pid != 0) {
+    } else 
+    #endif
+    if (pid != 0) {
         ::kill(pid, SIGINT);
     }
 }
@@ -254,6 +260,7 @@ void GDB::processLine(const QByteArray&
 
                emit internalCommandOutput(QString::fromUtf8(line) + '\n');
 
+               #if defined(Q_OS_MAC)
                if (result.reason == "thread-group-started") {
                    receivedReply_ = true;
                    //     (gdb) -exec-run
@@ -270,6 +277,7 @@ void GDB::processLine(const QByteArray&
                        }
                    }
                }
+	       #endif
 
                // FIXME: the code below should be reviewed to consider result record
                // subtype when doing all decisions.
diff -rupN a/debuggers/gdb/gdb.h b/debuggers/gdb/gdb.h
--- a/debuggers/gdb/gdb.h	2015-02-02 09:19:46.000000000 -0500
+++ b/debuggers/gdb/gdb.h	2015-08-03 18:14:34.000000000 -0400
@@ -144,7 +144,9 @@ private:
     
     bool receivedReply_;
     bool isRunning_;
+    #if defined(Q_OS_MAC)
     unsigned long childPid_;
+    #endif
 };
 }
 
