diff --git a/src/manager/CMakeLists.txt b/src/manager/CMakeLists.txt
index a432f32..ae00bc6 100644
--- a/src/manager/CMakeLists.txt
+++ b/src/manager/CMakeLists.txt
@@ -1,3 +1,5 @@
+add_definitions(-DTRANSLATION_DOMAIN=\"kcmkwallet\")
+
 ########### next target ###############
 set(kwalletmanager_SRCS 
     kwalletmanager.cpp
@@ -36,7 +38,10 @@ ki18n_wrap_ui(kwalletmanager_SRCS
 file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/*-apps-kwalletmanager.png")
 ecm_add_app_icon(kwalletmanager_SRCS ICONS ${ICONS_SRCS})
 
+
+
 add_executable(kwalletmanager5 ${kwalletmanager_SRCS})
+add_executable(kwalletmanager5_kauthdummy kauthdummy.cpp)
 
 target_link_libraries(kwalletmanager5
     Qt5::Core
@@ -52,7 +57,13 @@ target_link_libraries(kwalletmanager5
     KF5::KDELibs4Support
     KF5::Archive
 )
+target_link_libraries(kwalletmanager5_kauthdummy
+    Qt5::Core
+    KF5::CoreAddons
+    KF5::Auth
+)
 
+install(TARGETS kwalletmanager5_kauthdummy DESTINATION ${KAUTH_HELPER_INSTALL_DIR} )
 install(TARGETS kwalletmanager5  ${INSTALL_TARGETS_DEFAULT_ARGS} )
 install(FILES kwalletmanager.rc
         DESTINATION ${KXMLGUI_INSTALL_DIR}/kwalletmanager5)
@@ -70,3 +81,8 @@ ecm_install_icons( ICONS
 64-apps-kwalletmanager2.png
 64-apps-kwalletmanager.png
 DESTINATION ${ICON_INSTALL_DIR} )
+
+kauth_install_actions(org.kde.kwalletmanager5 kwalletmanager.actions)
+kauth_install_helper_files(kwalletmanager5_kauthdummy, org.kde.kwalletmanager5 root)
+
+
diff --git a/src/manager/kauthdummy.cpp b/src/manager/kauthdummy.cpp
new file mode 100644
index 0000000..da9ad1f
--- /dev/null
+++ b/src/manager/kauthdummy.cpp
@@ -0,0 +1,32 @@
+#include "kauthdummy.hpp"
+
+ActionReply KAuthDummy::viewdata(QVariantMap args)
+{
+	ActionReply reply;
+	(void) args;
+	return reply;
+}
+
+ActionReply KAuthDummy::modifydata(QVariantMap args)
+{
+	ActionReply reply;
+	(void) args;
+	return reply;
+}
+
+ActionReply KAuthDummy::changepassword(QVariantMap args)
+{
+	ActionReply reply;
+	(void) args;
+	return reply;
+}
+
+ActionReply KAuthDummy::exportwallet(QVariantMap args)
+{
+	ActionReply reply;
+	(void) args;
+	return reply;
+}
+
+KAUTH_HELPER_MAIN("org.kde.kwalletmanager5", KAuthDummy)
+
diff --git a/src/manager/kauthdummy.hpp b/src/manager/kauthdummy.hpp
new file mode 100644
index 0000000..4a134fb
--- /dev/null
+++ b/src/manager/kauthdummy.hpp
@@ -0,0 +1,14 @@
+#include <kauth.h>
+
+using namespace KAuth;
+
+class KAuthDummy : public QObject
+{
+    Q_OBJECT
+
+    public slots:
+        ActionReply viewdata(QVariantMap args);
+	ActionReply modifydata(QVariantMap args);
+	ActionReply changepassword(QVariantMap args);
+	ActionReply exportwallet(QVariantMap args);
+};
diff --git a/src/manager/kwalleteditor.cpp b/src/manager/kwalleteditor.cpp
index de239f2..0cc6fa9 100644
--- a/src/manager/kwalleteditor.cpp
+++ b/src/manager/kwalleteditor.cpp
@@ -42,6 +42,7 @@
 #include <ktoolbar.h>
 #include <QIcon>
 #include <KTreeWidgetSearchLine>
+#include <kauth.h>
 
 #include <QCheckBox>
 #include <QClipboard>
@@ -438,6 +439,13 @@ void KWalletEditor::createFolder()
 
 void KWalletEditor::saveEntry()
 {
+    KAuth::Action viewAction("org.kde.kwalletmanager5.modifydata");
+    viewAction.setHelperId("org.kde.kwalletmanager5");
+    auto job = viewAction.execute();
+    job->exec();
+    if (job->error())
+            return;
+
     int rc = 1;
     QTreeWidgetItem *item = _displayedItem; //  _entryList->currentItem();
     _saveChanges->setEnabled(false);
@@ -543,6 +551,7 @@ void KWalletEditor::entrySelectionChanged(QTreeWidgetItem *item)
                 }
             } else if (ci->entryType() == KWallet::Wallet::Map) {
                 _entryStack->setCurrentIndex(2);
+                showHideMapEditorValue(false);
                 if (_w->readMap(item->text(0), _currentMap) == 0) {
                     _mapEditor->reload();
                     _entryName->setText(i18n("Name-Value Map: %1", item->text(0)));
@@ -727,6 +736,13 @@ void KWalletEditor::listContextMenuRequested(const QPoint &pos)
 
 void KWalletEditor::copyPassword()
 {
+    KAuth::Action viewAction("org.kde.kwalletmanager5.viewdata");
+    viewAction.setHelperId("org.kde.kwalletmanager5");
+    auto job = viewAction.execute();
+    job->exec();
+    if (job->error())
+        return;
+
     QTreeWidgetItem *item = _entryList->currentItem();
     if (_w && item) {
         QString pass;
@@ -738,12 +754,15 @@ void KWalletEditor::copyPassword()
 
 void KWalletEditor::newEntry()
 {
+    KAuth::Action viewAction("org.kde.kwalletmanager5.modifydata");
     QTreeWidgetItem *item = _entryList->currentItem();
-    QString n;
+    QString n = "";
     bool ok;
 
     KWalletFolderItem *fi;
 
+    viewAction.setHelperId("org.kde.kwalletmanager5");
+
     //set the folder where we're trying to create the new entry
     if (_w && item) {
         QTreeWidgetItem *p = item;
@@ -762,13 +781,21 @@ void KWalletEditor::newEntry()
     do {
         n = QInputDialog::getText(this, i18n("New Entry"),
                                   i18n("Please choose a name for the new entry:"),
-                                  QLineEdit::Normal, QString(),
+                                  QLineEdit::Normal, n,
                                   &ok);
 
         if (!ok) {
             return;
         }
 
+        // Authenticate the user and if the password prompt
+        // is closed without authenticating show the dialog again
+        // so that the entry name is not lost
+        auto job = viewAction.execute();
+        job->exec();
+        if (job->error())
+            continue;
+
         // FIXME: prohibits the use of the subheadings
         if (fi->contains(n)) {
             int rc = KMessageBox::questionYesNo(this, i18n("Sorry, that entry already exists. Try again?"), QString(), KGuiItem(i18n("Try Again")), KGuiItem(i18n("Do Not Try")));
@@ -845,6 +872,15 @@ void KWalletEditor::listItemChanged(QTreeWidgetItem *item, int column)
             return;
         }
 
+        KAuth::Action viewAction("org.kde.kwalletmanager5.modifydata");
+        viewAction.setHelperId("org.kde.kwalletmanager5");
+        auto job = viewAction.execute();
+        job->exec();
+        if (job->error()) {
+            i->restoreName();
+            return;
+        }
+
         if (_w->renameEntry(i->name(), t) == 0) {
             i->setName(t);
             KWalletContainerItem *ci = dynamic_cast<KWalletContainerItem *>(item->parent());
@@ -871,6 +907,13 @@ void KWalletEditor::deleteEntry()
     if (_w && item) {
         int rc = KMessageBox::warningContinueCancel(this, i18n("Are you sure you wish to delete the item '%1'?", item->text(0)), QString(), KStandardGuiItem::del());
         if (rc == KMessageBox::Continue) {
+            KAuth::Action viewAction("org.kde.kwalletmanager5.modifydata");
+            viewAction.setHelperId("org.kde.kwalletmanager5");
+            auto job = viewAction.execute();
+	    job->exec();
+            if (job->error())
+                return;
+
             KWalletFolderItem *fi = dynamic_cast<KWalletFolderItem *>(item->parent()->parent());
             if (!fi) {
                 KMessageBox::error(this, i18n("An unexpected error occurred trying to delete the entry"));
@@ -913,13 +956,31 @@ void KWalletEditor::hidePasswordContents()
 
 void KWalletEditor::showPasswordContents()
 {
-    _entryStack->setCurrentIndex(1);
+    KAuth::Action viewAction("org.kde.kwalletmanager5.viewdata");
+    viewAction.setHelperId("org.kde.kwalletmanager5");
+    auto job = viewAction.execute();
+    job->exec();
+    if (!job->error()) {
+        _entryStack->setCurrentIndex(1);
+    } else {
+        _alwaysShowContents = false;
+        _alwaysShowContentsAction->setChecked(false);
+        KMessageBox::sorry(this, i18n("%1", Qt::escape(job->errorText())));
+    }
 }
 
 void KWalletEditor::showHideMapEditorValue(bool show)
 {
     if (show) {
-        _mapEditor->showColumn(2);
+        KAuth::Action viewAction("org.kde.kwalletmanager5.viewdata");
+        viewAction.setHelperId("org.kde.kwalletmanager5");
+        auto job = viewAction.execute();
+	job->exec();
+        if (!job->error()) {
+            _mapEditor->showColumn(2);
+        } else {
+            _mapEditorShowHide->setChecked(false);
+        }
     } else {
         _mapEditor->hideColumn(2);
     }
@@ -941,6 +1002,13 @@ void KWalletEditor::importWallet()
         return;
     }
 
+    KAuth::Action viewAction("org.kde.kwalletmanager5.modifydata");
+    viewAction.setHelperId("org.kde.kwalletmanager5");
+    auto job = viewAction.execute();
+    job->exec();
+    if (job->error())
+        return;
+
     KWallet::Wallet *w = KWallet::Wallet::openWallet(tmpFile, effectiveWinId(), KWallet::Wallet::Path);
     if (w && w->isOpen()) {
         MergePlan mp = Prompt;
@@ -1098,6 +1166,13 @@ void KWalletEditor::importXML()
         return;
     }
 
+    KAuth::Action viewAction("org.kde.kwalletmanager5.modifydata");
+    viewAction.setHelperId("org.kde.kwalletmanager5");
+    auto job = viewAction.execute();
+    job->exec();
+    if (job->error())
+        return;
+
     QDomNode n = top.firstChild();
     MergePlan mp = Prompt;
     while (!n.isNull()) {
@@ -1172,6 +1247,17 @@ void KWalletEditor::importXML()
 
 void KWalletEditor::exportXML()
 {
+    QUrl qurl = QFileDialog::getSaveFileUrl(this, QString(), QUrl(), QLatin1String("*.xml"));
+    if (qurl.isEmpty())
+        return;
+
+    KAuth::Action viewAction("org.kde.kwalletmanager5.exportwallet");
+    viewAction.setHelperId("org.kde.kwalletmanager5");
+    auto job = viewAction.execute();
+    job->exec();
+    if (job->error())
+        return;
+
     QTemporaryFile tf;
     tf.open();
     QXmlStreamWriter xml(&tf);
diff --git a/src/manager/kwalletmanager.actions b/src/manager/kwalletmanager.actions
new file mode 100644
index 0000000..c482b73
--- /dev/null
+++ b/src/manager/kwalletmanager.actions
@@ -0,0 +1,79 @@
+[Domain]
+Name=KDE Wallet system
+Name[bs]=Sistem KDE novčanika
+Name[ca]=Sistema de carteres del KDE
+Name[ca@valencia]=Sistema de cartera del KDE
+Name[cs]=Systém úschovny pro KDE
+Name[da]=KDE's tegnebogssystem
+Name[de]=Passwortspeicher für KDE
+Name[el]=Σύστημα πορτοφολιού του KDE
+Name[en_GB]=KDE Wallet system
+Name[es]=Sistema de carteras de KDE
+Name[et]=KDE turvalaekasüsteem
+Name[eu]=KDEren zorro sistema
+Name[fi]=KDE:n lompakkojärjestelmä
+Name[fr]=Systèmes de portefeuilles de KDE
+Name[gl]=Sistema de carteiras de KDE
+Name[hu]=KDE jelszókezelő rendszer
+Name[ia]=KDE Wallet  System (Systema de portafolio de KDE)
+Name[it]=Il sistema di portafogli di KDE
+Name[kk]=KDE әмиян жүйесі
+Name[ko]=KDE 지갑 시스템
+Name[nb]=KDEs Lommebok-system
+Name[nds]=KDE-Knippensysteem
+Name[nl]=KDE-portefeuillesysteem
+Name[pa]=KDE ਵਾਲਿਟ ਸਿਸਟਮ
+Name[pl]=System portfela KDE
+Name[pt]=Sistema da Carteira do KDE
+Name[pt_BR]=Sistema de carteiras do KDE
+Name[ro]=Sistemul de portofele KDE
+Name[ru]=Бумажник KDE
+Name[sk]=Systém KDE peňaženky
+Name[sl]=Sistem listnic za KDE
+Name[sr]=КДЕ систем новчаника
+Name[sr@ijekavian]=КДЕ систем новчаника
+Name[sr@ijekavianlatin]=KDE sistem novčanika
+Name[sr@latin]=KDE sistem novčanika
+Name[sv]=KDE:s plånbokssystem
+Name[tr]=KDE Cüzdan Sistemi
+Name[uk]=Система роботи з буферами даних KDE
+Name[x-test]=xxKDE Wallet systemxx
+Name[zh_CN]=KDE 钱包系统
+Name[zh_TW]=KDE 錢包系統
+Icon=kwalletmanager
+
+[org.kde.kwalletmanager5.viewdata]
+Name=View saved data
+Name[en_GB]=View saved data
+Name[es]=Ver información guardada en la cartera
+Description=Allow viewing of data saved in wallet
+Description[en_GB]=Allow viewing of data saved in wallet
+Description[es]=Permitir ver la información guardada en la cartera
+Policy=auth_self
+
+[org.kde.kwalletmanager5.modifydata]
+Name=Modify saved data
+Name[en_GB]=Modify saved data
+Name[es]=Modificar información guardada en la cartera
+Description=Allow modification of data saved in wallet
+Description[en_GB]=Allow modification of data saved in wallet
+Description[es]=Permitir modificar la información guardada en la cartera
+Policy=auth_self
+
+[org.kde.kwalletmanager5.changepass]
+Name=Change wallet password
+Name[en_GB]=Change wallet password
+Name[es]=Cambiar la contraseña de la cartera
+Description=Allow changing wallet password
+Description[en_GB]=Allow changing wallet password
+Description[es]=Permitir cambiar la contraseña de la cartera
+Policy=auth_self
+
+[org.kde.kwalletmanager5.exportwallet]
+Name=Export wallet
+Name[en_GB]=Export wallet
+Name[es]=Exportar cartera
+Description=Allow export wallet
+Description[en_GB]=Allow export wallet
+Description[es]=Permitir exportar la cartera
+Policy=auth_self
diff --git a/src/manager/kwalletmanager.cpp b/src/manager/kwalletmanager.cpp
index 64a90dd..9ee34ca 100644
--- a/src/manager/kwalletmanager.cpp
+++ b/src/manager/kwalletmanager.cpp
@@ -48,6 +48,7 @@
 #include <QRegExpValidator>
 #include <QTimer>
 #include <QFileDialog>
+#include <kauth.h>
 
 
 KWalletManager::KWalletManager(QWidget *parent, const char *name, Qt::WFlags f)
@@ -353,6 +354,13 @@ void KWalletManager::deleteWallet()
     if (rc != KMessageBox::Continue) {
         return;
     }
+    KAuth::Action viewAction("org.kde.kwalletmanager5.modifydata");
+    viewAction.setHelperId("org.kde.kwalletmanager5");
+    auto job = viewAction.execute();
+    job->exec();
+    if (job->error()) {
+        return;
+    }
     rc = KWallet::Wallet::deleteWallet(walletName);
     if (rc != 0) {
         KMessageBox::sorry(this, i18n("Unable to delete the wallet. Error code was %1.", rc));
diff --git a/src/manager/walletcontrolwidget.cpp b/src/manager/walletcontrolwidget.cpp
index dc47fff..b5fd327 100644
--- a/src/manager/walletcontrolwidget.cpp
+++ b/src/manager/walletcontrolwidget.cpp
@@ -33,6 +33,7 @@
 #include <QPropertyAnimation>
 #include <QTimer>
 #include <QToolButton>
+#include <kauth.h>
 
 WalletControlWidget::WalletControlWidget(QWidget *parent, const QString &walletName):
     QWidget(parent),
@@ -163,7 +164,12 @@ void WalletControlWidget::onDisconnectApplication()
 
 void WalletControlWidget::onChangePassword()
 {
-    KWallet::Wallet::changePassword(_walletName, effectiveWinId());
+	KAuth::Action viewAction("org.kde.kwalletmanager5.changepass");
+	viewAction.setHelperId("org.kde.kwalletmanager5");
+	auto job = viewAction.execute();
+	job->exec();
+	if (!job->error())
+		KWallet::Wallet::changePassword(_walletName, effectiveWinId());
 }
 
 void WalletControlWidget::hideEvent(QHideEvent *)
