diff -rupN a/src/manager/CMakeLists.txt b/src/manager/CMakeLists.txt
--- a/src/manager/CMakeLists.txt	2014-10-09 03:07:50.000000000 -0400
+++ b/src/manager/CMakeLists.txt	2015-07-22 21:43:28.000000000 -0400
@@ -35,3 +35,4 @@ install(TARGETS kwalletmanager  ${INSTAL
 install( FILES kwalletmanager.rc DESTINATION ${DATA_INSTALL_DIR}/kwalletmanager)
 
 kde4_install_icons( ${ICON_INSTALL_DIR} )
+kde4_install_auth_actions(org.kde.kwallet kwalletmanager.actions)
diff -rupN a/src/manager/kwalleteditor.cpp b/src/manager/kwalleteditor.cpp
--- a/src/manager/kwalleteditor.cpp	2014-10-09 03:07:50.000000000 -0400
+++ b/src/manager/kwalleteditor.cpp	2015-07-22 22:02:34.000000000 -0400
@@ -61,6 +61,7 @@
 #include <kicon.h>
 #include <KAction>
 #include <KTreeWidgetSearchLine>
+#include <kauthaction.h>
 
 QAction *KWalletEditor::_newFolderAction =0;
 QAction *KWalletEditor::_deleteFolderAction =0;
@@ -929,7 +930,10 @@ void KWalletEditor::hidePasswordContents
 
 
 void KWalletEditor::showPasswordContents() {
-	_entryStack->setCurrentIndex(1);
+	KAuth::Action viewAction("org.kde.kwallet.viewpass");
+	KAuth::ActionReply reply = viewAction.execute();
+	if (!reply.failed())
+		_entryStack->setCurrentIndex(1);
 }
 
 
@@ -1186,6 +1190,11 @@ void KWalletEditor::importXML() {
 
 
 void KWalletEditor::exportXML() {
+	KAuth::Action viewAction("org.kde.kwallet.exportwallet");
+	KAuth::ActionReply reply = viewAction.execute();
+	if (reply.failed())
+		return;
+
 	KTemporaryFile tf;
 	tf.open();
 	QXmlStreamWriter xml(&tf);
diff -rupN a/src/manager/kwalletmanager.actions b/src/manager/kwalletmanager.actions
--- a/src/manager/kwalletmanager.actions	1969-12-31 19:00:00.000000000 -0500
+++ b/src/manager/kwalletmanager.actions	2015-07-22 22:11:30.000000000 -0400
@@ -0,0 +1,70 @@
+[Domain]
+Name=KDE Wallet system
+Name[bs]=Sistem KDE novčanika
+Name[ca]=Sistema de carteres del KDE
+Name[ca@valencia]=Sistema de cartera del KDE
+Name[cs]=Systém úschovny pro KDE
+Name[da]=KDE's tegnebogssystem
+Name[de]=Passwortspeicher für KDE
+Name[el]=Σύστημα πορτοφολιού του KDE
+Name[en_GB]=KDE Wallet system
+Name[es]=Sistema de carteras de KDE
+Name[et]=KDE turvalaekasüsteem
+Name[eu]=KDEren zorro sistema
+Name[fi]=KDE:n lompakkojärjestelmä
+Name[fr]=Systèmes de portefeuilles de KDE
+Name[gl]=Sistema de carteiras de KDE
+Name[hu]=KDE jelszókezelő rendszer
+Name[ia]=KDE Wallet  System (Systema de portafolio de KDE)
+Name[it]=Il sistema di portafogli di KDE
+Name[kk]=KDE әмиян жүйесі
+Name[ko]=KDE 지갑 시스템
+Name[nb]=KDEs Lommebok-system
+Name[nds]=KDE-Knippensysteem
+Name[nl]=KDE-portefeuillesysteem
+Name[pa]=KDE ਵਾਲਿਟ ਸਿਸਟਮ
+Name[pl]=System portfela KDE
+Name[pt]=Sistema da Carteira do KDE
+Name[pt_BR]=Sistema de carteiras do KDE
+Name[ro]=Sistemul de portofele KDE
+Name[ru]=Бумажник KDE
+Name[sk]=Systém KDE peňaženky
+Name[sl]=Sistem listnic za KDE
+Name[sr]=КДЕ систем новчаника
+Name[sr@ijekavian]=КДЕ систем новчаника
+Name[sr@ijekavianlatin]=KDE sistem novčanika
+Name[sr@latin]=KDE sistem novčanika
+Name[sv]=KDE:s plånbokssystem
+Name[tr]=KDE Cüzdan Sistemi
+Name[uk]=Система роботи з буферами даних KDE
+Name[x-test]=xxKDE Wallet systemxx
+Name[zh_CN]=KDE 钱包系统
+Name[zh_TW]=KDE 錢包系統
+Icon=kwalletmanager
+
+[org.kde.kwallet.viewpass]
+Name=View saved passwords
+Name[en_GB]=View saved passwords
+Name[es]=Ver contraseñas en la cartera
+Description=Allow viewing saved passwords
+Description[en_GB]=Allow viewing saved passwords
+Description[es]=Permitir que se vean las contraseñas en la cartera
+Policy=auth_self
+
+[org.kde.kwallet.changepass]
+Name=Change wallet password
+Name[en_GB]=Change wallet password
+Name[es]=Cambiar la contraseña de la cartera
+Description=Allow changing wallet password
+Description[en_GB]=Allow changing wallet password
+Description[es]=Permitir cambiar la contraseña de la cartera
+Policy=auth_self
+
+[org.kde.kwallet.exportwallet]
+Name=Export wallet
+Name[en_GB]=Export wallet
+Name[es]=Exportar cartera
+Description=Allow export wallet
+Description[en_GB]=Allow export wallet
+Description[es]=Permitir exportar la cartera
+Policy=auth_self
diff -rupN a/src/manager/walletcontrolwidget.cpp b/src/manager/walletcontrolwidget.cpp
--- a/src/manager/walletcontrolwidget.cpp	2014-10-09 03:07:50.000000000 -0400
+++ b/src/manager/walletcontrolwidget.cpp	2015-07-22 21:45:49.000000000 -0400
@@ -31,6 +31,7 @@
 #include <kmenu.h>
 #include <KTabWidget>
 #include <kdebug.h>
+#include <kauthaction.h>
 
 WalletControlWidget::WalletControlWidget(QWidget* parent, const QString& walletName):
     QWidget(parent),
@@ -161,7 +162,10 @@ void WalletControlWidget::onDisconnectAp
 
 void WalletControlWidget::onChangePassword()
 {
-    KWallet::Wallet::changePassword(_walletName, effectiveWinId());
+	KAuth::Action viewAction("org.kde.kwallet.changepass");
+	KAuth::ActionReply reply = viewAction.execute();
+	if (!reply.failed())
+		KWallet::Wallet::changePassword(_walletName, effectiveWinId());
 }
 
 void WalletControlWidget::hideEvent(QHideEvent* )
